src/App.jsx

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import './App.css';
+import { GoodsList } from './components/GoodsList';
+import { Start } from './components/Start';
+
+const goodsFromServer = [
+  'Dumplings',
+  'Carrot',
+  'Eggs',
+  'Ice cream',
+  'Apple',
+  'Bread',
+  'Fish',
+  'Honey',
+  'Jam',
+  'Garlic',
+];
+
+class App extends React.Component {
+  state = {
+    isStartButtonVisible: true,
+  }
+
+  showGoodsList = () => {
+    this.setState({
+      isStartButtonVisible: false,
+    });
+  }
+
+  render() {
+    const { isStartButtonVisible } = this.state;
+
+    return (
+      <div className="app">
+        {isStartButtonVisible
+          ? <Start showGoodsList={this.showGoodsList} />
+          : <GoodsList goods={goodsFromServer} />
```

Comment:
Let's split the logic. Make all operations here and pass already transformed goods from server to the 'GoodsList' component